eclipse-theia/theia
# Add breadcrumbs for editors

This PR updates #6731 and completes the remaining to-do's from that PR.

From that description:
> This commit adds a breadcrumbs bar to the editor widget. It shows the path to the current file and outline information as breadcrumbs. A click of breadcrumbs allows to switch to siblings.

> This fixes #5475.

> How to test:
> Open an editor shows the breadcrumbs on top of the editor. Folders with files are clickable. A popup allows to select a file that will be opened immediately. Outline information is added to the breadcrumbs as well.

![image](https://user-images.githubusercontent.com/62660806/129743733-50a18d68-dd0d-4c5a-b901-664f84bbee65.png)


A few more details:
1. You should see breadcrumbs for `NavigatableWidget`s (mainly editors) that are docked in an area with a horizontal tabbar. Those breadcrumbs should contain at least the path to the file.
2. For editors with language support, you should get the content of the outline view for the location of your cursor.
  > Try refreshing when you have a TS file open. The layout should be restored. There were changes to the Outline view that required changes to its state restoration.
3. If you open a non-navigatable widget, the breadcrumbs bar should disappear.
4. There is a preference to disable breadcrumbs. If you disable the breadcrumbs, they should disappear, and they shouldn't reappear anywhere.
  > Watch out for any odd styling behavior in the transition from breadcrumbs <-> no breadcrumbs. I think I've handled it, but there may still be some quirks.
5. If you click a breadcrumb, it should open a popup. The popup should show other possibilities at the same level of organization (sibling folders, object fields, etc.).
6. You should be able to navigate the tree

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
